### DISCUSSION TITLE
How can I run a mina node on mobile phone

### DISCUSSION THREAD

**Discussion by JasonCeng** [MinaProtocol/mina#12584] [Q&A]
How can I run a mina node on mobile phone?How do I become a verifier and how do I become a snarker? If someone could let us know how to do all this that would be great. I fully believe this project will change crypto

**Comment by Sventimir** [comment]
I don't think it's possible at the moment. The node is being developed on Linux and it's also possible to run it on MacOS as far as I know. Making it run on Android would require substantial adaptations. I don't know about iOS, but I guess some (other) adaptations would be required as well. Although in principle nothing is stopping us from developing mobile apps for Mina node, the technology that has been chosen for the project does not play well with mobile devices, so adapting the code that we currently have to run on moblie devices would be a huge task. Additionally, a graphical client app would also be needed so that user could interact with their node. So, while it's doable, I doubt it's worth the effort at the moment.

There is a project being developed somewhere aiming at running the node in a web browser. That one should also be able to run on mobile devices.

**Comment by kaozenn** [comment]
On top of @Sventimir's comment, Mina currently only runs on AMD64 processors. Most mobile phones use ARM64 processors and the node will not start. We have not made it a priority yet, but we will address it soon, as Mac M1 also uses ARM64 processors.

**Comment by kaozenn** [comment]
@JasonCeng This may interest you: 
https://medium.com/openmina/introducing-the-web-node-an-in-browser-mina-node-that-verifies-blocks-and-transfers-funds-ebc59a57e79a
